### PR TITLE
Fix coverity issues (Apr 2023)

### PR DESCRIPTION
**Pull request by jantonguirao**
## Category:

**Other**

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/imgcodec/decoders/nvjpeg/nvjpeg.cc [hunk 1]
```diff
@@ -66,15 +66,15 @@ NvJpegDecoderInstance(int device_id, const std::map<std::string, any> &params)
 
   if (host_memory_padding_ > 0) {
     CUDA_CALL(nvjpegSetPinnedMemoryPadding(host_memory_padding_, nvjpeg_handle_));
-    for (auto thread_id : tp_->GetThreadIds()) {
+    for (const auto& thread_id : tp_->GetThreadIds()) {
```
Comment (JanuszL) [inline dali/imgcodec/decoders/nvjpeg/nvjpeg.cc:69]: I thought that `thread_id` is rather a lightweight object (a number).

Comment (jantonguirao) [reply dali/imgcodec/decoders/nvjpeg/nvjpeg.cc:69]: You are right. It was marked as fixes submitted but the fix was not submitted, that's why I did it. Let me revert

File: dali/operators/reader/loader/fits_loader.cc [hunk 2]
```diff
@@ -77,7 +77,7 @@ void FitsLoader::ReadSample(FitsFileWrapper& target) {
     target.data[output_idx].SetMeta(meta);
 
     // set file path
-    target.filename = std::move(path);
+    target.filename = path;
```
Comment (klecki) [inline dali/operators/reader/loader/fits_loader.cc:80]: What was wrong with the move here?

Comment (JanuszL) [reply dali/operators/reader/loader/fits_loader.cc:80]: Probably complier would do that for you and it may be considered redundant.

Comment (klecki) [reply dali/operators/reader/loader/fits_loader.cc:80]: Why would it, if there are valid copy and move assignments operators, the one that is selected would depend on the type of the right operand.

Comment (jantonguirao) [reply dali/operators/reader/loader/fits_loader.cc:80]: it's done in a loop, so we can use after move, that was the problem

**Review by JanuszL — APPROVED** [review]

File: dali/operators/reader/loader/fits_loader.cc (hunk 2 above)
Comment (jantonguirao) [reply dali/operators/reader/loader/fits_loader.cc:80]: Now that I look at it, I think it should be moved out of the loop. There's a single `target.filename`, but several `target.data[output_idx]`. We don't need to assign `target.filename` several times.

**Review by klecki — APPROVED** [review]

**Pull request merged**